code/script.js

```diff
@@ -229,97 +237,171 @@ const setSecret = () => {
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  } if (category === 'accessories') {
+    if (keep) {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      Yes, the person wears ${value}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    
+    } else {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person doesn't wear ${value}!`
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  } else if (category === 'other') {
+    if (keep) {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      Yes, the person is a ${value}!`
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    
+    } else {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person is not a ${value}!`
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    }
+  }  
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+  // When the user clicks on close (x), the modal will close
+  span.onclick = function() {
+   modal.style.display = "none";
+  }
+
+  // When the user clicks anywhere outside of the modal, close it
+  window.onclick = function(event) {
+    if (event.target == modal) {
+      modal.style.display = "none";
+    }
+  }
 
+  setTimeout(() => generateBoard(), 500);      
+} 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  let guessedPerson = personToConfirm  //storing the interaction from the player in a variable.
+  console.log(guessedPerson)
+  let confirmGuess = confirm(`Are you sure you want to guess on ${personToConfirm}?`)
+  if(confirmGuess) { //invoke the checkMyGuess function.
+    checkMyGuess(guessedPerson)
+  } 
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+const checkMyGuess = (personToCheck) => {   // 1. Check if the personToCheck is the same as the secret person's name
+  if ( personToCheck === secret.name ) {
+    console.log('correct')
```

Comment:
It is great that you're using console logs, they are really great way to understand code better :) But remember to delete them when you're done with the project :)